src/frontend/src/components/tables/InvenTreeTable.tsx

```diff
@@ -0,0 +1,347 @@
+    enableRefresh?: boolean;
+    pageSize?: number;
+    printingActions?: any[];
+    barcodeActions?: any[];
+    customActionGroups?: any[];
+    customFilters?: TableFilter[];
+}) {
+
+    // Check if any columns are switchable (can be hidden)
+    const hasSwitchableColumns = columns.some((col: any) => col.switchable);
+
+    // Check if custom filtering is enabled for this table
+    const hasCustomFilters = enableFilters && customFilters.length > 0;
+
+    // Pagination
+    const [page, setPage] = useState(1);
+
+    // Filter list visibility
+    const [filtersVisible, setFiltersVisible] = useState<boolean>(false);
+
+    // Map of currently active filters, {name: value}
+    const [activeFilters, setActiveFilters] = useState<any>({});
+
+    // Data columns
+    const [dataColumns, setDataColumns] = useState<any[]>(columns);
+
+    // List of hidden columns
+    let hiddenColumns = [];
+
+    // Search term
+    const [searchTerm, setSearchTerm] = useState<string>('');
+
+    let latestSearchTerm = '';
+
+    useEffect(() => {
+        // Keep a shadow copy of the state variable, so that we can update it asynchronously
+        latestSearchTerm = searchTerm;
+
+        loadHiddenColumns();
+        loadActiveFilters();
+    }, [searchTerm]);
+
+    // Load list of hidden columns from local storage
+    function loadHiddenColumns() {
+        hiddenColumns = JSON.parse(localStorage.getItem(`hidden-table-columns-${tableKey}`) || '[]');
+
+        // Update column visibility
+        let newColumns = [...dataColumns];
+
+        for (let idx = 0; idx < newColumns.length; idx++) {
+            newColumns[idx].hidden = hiddenColumns.includes(newColumns[idx].accessor);
+        }
+
+        setDataColumns(newColumns);
+    }
+
+    // Load list of active filters from local storage
+    function loadActiveFilters() {
+        // TODO
+    }
+
+    // Callback when search term is updated
+    function updateSearchTerm(term: string) {
+        term = term.trim();
+        // Ignore identical search terms
+        if (term == searchTerm) return;
+
+        setSearchTerm(term);
+        latestSearchTerm = term;
+        refetch();
+    }
+
+    // Data download callback
+    function downloadData(fileFormat: string) {
+        console.log("download data: " + fileFormat);
+    }
+
+    // Data Sorting
+    const [sortStatus, setSortStatus] = useState<DataTableSortStatus>({ columnAccessor: defaultSortColumn, direction: 'asc' });
+
+    // Return the ordering parameter
+    function getOrderingTerm() {
+        let key = sortStatus.columnAccessor;
+
+        // Sorting column not specified
+        if (key == '') {
+            return '';
+        }
+
+        // Find matching column:
+        // If column provides custom ordering term, use that
+        let column = dataColumns.find((col) => col.accessor == key);
+
+        if (column && column.ordering) {
+            return column.ordering;
+        } else {
+            return key;
+        }
```

Comment:
Just a tip, to avoid these long && chains you could use optional chaining (`column?.ordering`) and you could also combine that with the  or operator or even the nullish coalescing operator like this:
```suggestion
        return column?.ordering ||key;
```